crates/handlers/src/graphql/model/upstream_oauth.rs

```diff
@@ -45,6 +45,33 @@ impl UpstreamOAuth2Provider {
     pub async fn client_id(&self) -> &str {
         &self.provider.client_id
     }
+
+    /// The human-readable name of the provider.
+    pub async fn human_name(&self) -> Option<&str> {
+        self.provider.human_name.as_deref()
+    }
+
+    /// UpstreamOAuth2Links associated with this provider for the current user.
+    pub async fn upstream_oauth2_links_for_user(
+        &self,
+        ctx: &Context<'_>,
+    ) -> Result<Vec<UpstreamOAuth2Link>, async_graphql::Error> {
+        let state = ctx.state();
+        let user_id = ctx
+            .requester()
+            .user()
+            .ok_or_else(|| async_graphql::Error::new("User ID not found in the request context"))?
+            .id;
+
+        let mut repo = state.repository().await?;
+        let links = repo
+            .upstream_oauth_link()
+            .find_by_user_id(&self.provider, user_id)
+            .await?;
+        repo.cancel().await?;
```

Comment:
You technically can have multiple links for the same user and the same provider, which means this should be a list. You could be using `UpstreamOAuthLinkRepository::list` with the right filter? It's a little annoying because this API requires pagination, but on the other hand… we could just return the 100 first entries, and warn if (in a very unlikely scenario) there are more than this?

Reply:
Oh right. I missed that the list function allows filtering. I will look into that.

On the pagination yeah that was my thought as well here. Like how likely is it to have multiple links on the same provider. Arguably there should be only a single link per single provider. But not sure if its enforced to be like this.